File: web/api/provider.ts
import { api } from '@/lib/api'

// 获取提供商列表
export const getProviders = async (tenantId: string) => {
  try {
    const response = await api.get(`/providers?tenantId=${tenantId}`)
    return response.data || []
  } catch (error) {
    console.error('获取提供商列表失败', error)
    throw error
  }
}

// 获取提供商详情
export const getProviderById = async (providerId: string) => {
  try {
    const response = await api.get(`/providers/${providerId}`)
    return response.data
  } catch (error) {
    console.error('获取提供商详情失败', error)
    throw error
  }
}

// 添加提供商
export const createProvider = async (data: {
  tenantId: string
  providerName: string
  logo?: string
  apiKey: string
  endpoint?: string
}) => {
  try {
    const response = await api.post('/providers', data)
    return response.data
  } catch (error) {
    console.error('添加提供商失败', error)
    throw error
  }
}

// 更新提供商
export const updateProvider = async (
  providerId: string,
  data: {
    providerName?: string
    logo?: string
    apiKey?: string
    endpoint?: string
    status?: string
  },
) => {
  try {
    const response = await api.put(`/providers/${providerId}`, data)
    return response.data
  } catch (error) {
    console.error('更新提供商失败', error)
    throw error
  }
}

// 删除提供商
export const deleteProvider = async (providerId: string) => {
  try {
    const response = await api.delete(`/providers/${providerId}`)
    return response.data
  } catch (error) {
    console.error('删除提供商失败', error)
    throw error
  }
}

// 获取提供商下的模型列表
export const getModelsByProviderId = async (providerId: string) => {
  try {
    const response = await api.get(`/models/provider/${providerId}`)
    return response.data || []
  } catch (error) {
    console.error('获取模型列表失败', error)
    throw error
  }
}

export const providerService = {
  getProviders,
  getProviderById,
  createProvider,
  updateProvider,
  deleteProvider,
  getModelsByProviderId,
}

File: api/src/main/java/com/yxboot/modules/ai/entity/Provider.java
package com.yxboot.modules.ai.entity;

import java.time.LocalDateTime;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.yxboot.modules.ai.enums.CommonStatus;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

/**
 * 模型提供商实体类
 * 
 * @author Boya
 */
@Data
@TableName("provider")
@Schema(description = "模型提供商信息")
public class Provider {

    @TableId(value = "provider_id", type = IdType.ASSIGN_ID)
    @Schema(description = "提供商ID")
    private Long providerId;

    @Schema(description = "所属租户ID")
    private Long tenantId;

    @Schema(description = "提供商名称")
    private String providerName;

    @Schema(description = "Logo")
    private String logo;

    @Schema(description = "API密钥")
    private String apiKey;

    @Schema(description = "终端地址")
    private String endpoint;

    @Schema(description = "最后使用时间")
    private LocalDateTime lastUsedTime;

    @Schema(description = "状态")
    private CommonStatus status;

    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}